onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2843,12 +2812,12 @@ Status PadOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
   const auto* pads_initializer = model_builder.GetConstantInitializer(pads);
   ORT_RETURN_IF_NOT(pads_initializer, "pads must be a constant");
 
-  std::vector<uint8_t> pads_initializer_raw_data{};
-  ORT_RETURN_IF_ERROR(utils::UnpackInitializerData(*pads_initializer, pads_initializer_raw_data));
+  Initializer pads_initializer_raw_data(*pads_initializer);
+  auto pads_tensor_span = pads_initializer_raw_data.DataAsByteSpan<uint8_t>();
```

Comment:
General comment for these changes. Previously we only had the raw data, but now that we're using Initializer and it's converted to a Tensor we don't need to do everything using bytes. If we know something is of a specific data type it would be better to use `data<T>` and the number of elements instead of using the raw bytes/number of bytes.

Reply:
It's true. we can remove those redundant type-check code.